examples/abitest/module_0/rust/src/lib.rs

```diff
@@ -1624,6 +1628,41 @@ impl FrontendNode {
         Ok(())
     }
 
+    fn test_channel_wait_with_downgrade(&mut self) -> TestResult {
+        let test_msg = vec![0x01, 0x02, 0x03];
+
+        let (public_out_channel, public_in_channel) =
+            oak::channel_create("Public test channel", &Label::public_untrusted())
+                .expect("could not create channel");
+
+        // Test that [`oak::wait_on_channels_with_downgrade`] is successful and returns
+        // [`ChannelReadStatus::ReadReady`].
+        oak::channel_write(public_out_channel, &test_msg, &[]).expect("could not write to channel");
+        expect_eq!(
+            vec![ChannelReadStatus::ReadReady],
+            status_convert(oak::wait_on_channels_with_downgrade(&[public_in_channel]))?
+        );
+        expect_eq!(Ok(()), oak::channel_close(public_out_channel.handle));
+        expect_eq!(Ok(()), oak::channel_close(public_in_channel.handle));
+
+        let (private_out_channel, private_in_channel) =
+            oak::channel_create("Private test channel", &confidentiality_label(top()))
+                .expect("could not create channel");
+
+        // Test that [`oak::wait_on_channels_with_downgrade`] returns
+        // [`ChannelReadStatus::PermissionDenied`].
+        oak::channel_write(private_out_channel, &test_msg, &[])
+            .expect("could not write to channel");
+        expect_eq!(
+            vec![ChannelReadStatus::PermissionDenied],
+            status_convert(oak::wait_on_channels_with_downgrade(&[private_in_channel]))?
+        );
+        expect_eq!(Ok(()), oak::channel_close(private_out_channel.handle));
+        expect_eq!(Ok(()), oak::channel_close(private_in_channel.handle));
+
+        Ok(())
```

Comment:
Perhaps add a todo to check the case where it is different from the normal `wait_on_channels` once the ABI tests can support wasm hash labels.